components/ht16k33_alpha/display.py

```diff
@@ -1,25 +1,32 @@
 import esphome.codegen as cg
 import esphome.config_validation as cv
 from esphome.components import display, i2c
-from esphome.const import CONF_ID, CONF_LAMBDA
+from esphome.const import CONF_ID, CONF_LAMBDA, CONF_TYPE
 
 DEPENDENCIES = ['i2c']
 
-ht16k33_alpha_ns = cg.esphome_ns.namespace('ht16k33_alpha')
-HT16K33AlphaDisplay = ht16k33_alpha_ns.class_('HT16K33AlphaDisplay', cg.PollingComponent, i2c.I2CDevice)
+ht16k33_ns = cg.esphome_ns.namespace('ht16k33')
+HT16K33BaseDisplay = ht16k33_ns.class_('HT16K33BaseDisplay', cg.PollingComponent, i2c.I2CDevice)
+
+TYPES = {
+  "ALPHA": ht16k33_ns.class_("HT16K33AlphaDisplay", HT16K33BaseDisplay),
+  "7SEGMENT": ht16k33_ns.class_("HT16K337SegmentDisplay", HT16K33BaseDisplay),
+}
 
 CONF_SCROLL = "scroll"
 CONF_SCROLL_SPEED = "scroll_speed"
 CONF_SCROLL_DWELL = "scroll_dwell"
 CONF_SCROLL_DELAY = "scroll_delay"
 CONF_SECONDARY_DISPLAYS = "secondary_displays"
 
+
 CONFIG_SECONDARY = cv.Schema({
     cv.GenerateID(): cv.declare_id(i2c.I2CDevice)
 }).extend(i2c.i2c_device_schema(None))
 
 CONFIG_SCHEMA = display.BASIC_DISPLAY_SCHEMA.extend({
-    cv.GenerateID(): cv.declare_id(HT16K33AlphaDisplay),
+    cv.GenerateID(): cv.declare_id(HT16K33BaseDisplay),
+    cv.Optional(CONF_TYPE, default="ALPHA"): cv.enum(TYPES, upper=True),
```

Comment:
If we finally rename the component to just "ht16k33", it would probably make sense to make this a required parameter, since it would be a breaking change anyway.